firmware/scripts/format.py

```
@@ -0,0 +1,67 @@
+
+    if not files_to_format:
+        print("No C/C++ files found to format")
+        return
+
+    print("🎨 Formatting {} files with AStyle...".format(len(files_to_format)))
+    print("   Config: {}".format(astyle_config))
+
+    cmd = ["astyle", "--options={}".format(astyle_config)] + [str(f) for f in files_to_format]
+
+    try:
+        result = subprocess.run(cmd, capture_output = True, text = True)
```

Ruff (0.14.8)

`subprocess` call: check for execution of untrusted input

(S603)